### PR TITLE
{tools}[fosscuda/2020b] jax v0.2.19 w/ Python 3.8.6

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

I'm not sure if we also want to set `--cuda_version=` and `--cudnn_version=`.

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bber0501u03b.bb2.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/35078260f9bda72222c903830fb4cdf9 for a full test report.

File: easybuild/easyconfigs/j/jax/jax-0.2.19-fosscuda-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+local_jax_build_cmd = 'python build/build.py '
+local_jax_build_cmd += '--target_cpu_features=native '
+local_jax_build_cmd += '--bazel_startup_options="--output_user_root=%(builddir)s" '
+local_jax_build_cmd += '--bazel_path="$EBROOTBAZEL/bin/bazel" '
+local_jax_build_cmd += '--bazel_options=--subcommands '
+local_jax_build_cmd += '--bazel_options=--jobs=%(parallel)s --bazel_options=--action_env=PYTHONPATH '
+local_jax_build_cmd += '--bazel_options=--action_env=EBPYTHONPREFIXES '
+local_jax_build_cmd += '--enable_cuda --enable_nccl --cuda_compute_capabilities=%(cuda_compute_capabilities)s '
+local_jax_build_cmd += '--cuda_path=$EBROOTCUDA --cudnn_path=$EBROOTCUDNN '
```
Comment (boegel) [inline easybuild/easyconfigs/j/jax/jax-0.2.19-fosscuda-2020b.eb:45]: @branfosj Together with the stuff we have below, I think we should look into a custom easyblock for `jaxlib`, just to clean this stuff up a bit...

If you agree, I think that's something @deniskristak can look into

Comment (branfosj) [reply easybuild/easyconfigs/j/jax/jax-0.2.19-fosscuda-2020b.eb:45]: Yes an easyblock will be needed. We need to control when we set `--target_cpu_features=native `, as this adds `-march=native` and that does not work on POWER.

Comment (boegel) [reply easybuild/easyconfigs/j/jax/jax-0.2.19-fosscuda-2020b.eb:45]: We could use a runtime patch that changes the `-march=native` that gets injected based on which CPU architecture we're on?

**Comment by boegel** [comment]
@branfosj Hmm, interesting how this worked for you, but I'm hitting this:

```
ERROR: /tmp/easybuild_build/jax/0.2.19/fosscuda-2020b/ed092ae73b55be93f63253ba3147b947/external/org_tensorflow/tensorflow/core/util/BUILD:535:11: undeclared inclusion(s) in rule '@org_tensorflow//tensorflow/core/util:version_info':
this rule is missing dependency declarations for the following files included by 'external/org_tensorflow/tensorflow/core/util/version_info.cc':
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/stdarg.h'
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/stddef.h'
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/stdint.h'
Target //build:build_wheel failed to build
```

**Comment by boegel** [comment]
It gets better: it's failing in different ways in subsequent runs:

```
ERROR: /tmp/easybuild_build/jax/0.2.19/fosscuda-2020b/ed092ae73b55be93f63253ba3147b947/external/com_google_absl/absl/base/BUILD.bazel:83:11: undeclared inclusion(s) in rule '@com_google_absl//absl/base:spinlock_wait':
this rule is missing dependency declarations for the following files included by 'com_google_absl/absl/base/internal/spinlock_wait.cc':
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/stdint.h'
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/limits.h'
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/syslimits.h'
  '/software/GCCcore/10.2.0/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/stddef.h'
```

**Comment by boegel** [comment]
This looks like the kind of trouble that @Flamefire is very familiar with...

**Comment by branfosj** [comment]
My successful build was on a system with no GPUs. I can build it successfully on systems with GPUs, but the tests never seem to complete - 100% GPU use for hours.

**Comment by branfosj** [comment]
And, I've marked it as draft while we work further on it.

**Comment by boegel** [comment]
The errors I'm hitting are also discussed in https://github.com/google/jax/issues/4493, and it seems like it may have something to do with installing in a path that contains symbolic links, or using a compiler which is installed in a path with symbolic links (🙄).

**Comment by boegel** [comment]
We'll probably need something like we have in the TensorFlow easyblock: a function to patch CROSSTOOL files to ensure the right compiler paths are used (see [patch_crosstool_files](https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/t/tensorflow.py#L662))...

**Comment by Flamefire** [comment]
New version at https://github.com/easybuilders/easybuild-easyconfigs/pull/13760 with updated easyblock

**Pull request closed without merging**